text/0068-org-user-token-restriction.md

```diff
@@ -0,0 +1,42 @@
+- Start Date: 2023-01-18
+- RFC Type: feature
+- RFC PR: -
+- RFC Status: `draft`
+
+# Summary
+
+Provide an option to organization owners to disallow all user interaction with their organization that is using a user API token for authentication.
+
+# Motivation
+
+Sentry treats users and organizations as two separate entities. A user can be a part of multiple organizations.
+
+An organization may have stricter requirements or a lower risk appetite compared to a user. We provide options for organizations to require SSO and 2FA to be enabled on a user account that wishes to interact with their organization. However, a user can authenticate with a personal API token, which inherently does not require controls like a 2FA prompt.
+
+As a result, protecting the user API token lies solely on the user. Leaking secrets in source control, personal computers infected with malware, or vendor breaches can lead to these tokens being compromised, exposing an organization to increased risk in Sentry.
+
+If a personal user API token is leaked, an organization owner does not have the ability to revoke the token and must coordinate with the user to do so -- further increasing the time an organization is potentially exposed.
+
+# Solution
+
+1. Add a toggle in the Organization Settings to deny use of personal API tokens for authentication against their organization.
```

Comment:
I don't know if this is feasible or even useful, but more-generalized solution would be

```
1. Add a Organization Setting to limit the maximum lifetime of acceptable tokens. Existing organizations will default to "unlimited".
1. Update authentication middleware to compare token's lifetime versus the targeted organization's preference.
```

The boolean in this document is equivalent to setting maximum-lifetime to either `0s` or `unlimited`. 

For comparison, GCP access tokens are only valid for (up to) one hour.

Reply:
I believe this is out of scope. The issue isn't that an expired token is being used, it's that a user token is being used.

While a policy type system that limits the lifetime of a token would be more flexible, I haven't seen any asks for this. An on/off toggle is more inline with what I have seen requested by customers.

These tokens are unknown and outside of the control organization owners. This can unnecessarily and unknowingly expose an organization to increased risk if a user of their org accidentally leaks their token.